### PR TITLE
Kafka 0.9

### PR DESCRIPTION
**Pull request by rabbah**
Passed in playground 99.

1. fix typo in method name
2. remove deadcode classpath reference from build file
3. add cheerio package and move packages to nodejsAction Dockerfile
4. remove Kafka 0.8 references, upgrade to Kafka 0.9 and clean up imports
5. assign activations of triggers, rules to the same namespace as actions (subject default namespace)
6. add trigger payload to trigger activation
7. change type of quota exemptions from String to Set[Subject]
8. update Kafka tests and add new test for Kafka Connectors
9. in wsk CLI, separate reading HTTP response from JSON parsing so that if it possible to report an error from HTTP response, we can

In moving to Kafka 0.9, the deserialization of the raw message into an ActivationMessage is deferred to the Activator and Invoker so that the Kafka connector is agnostic of the message type. The streaming interface used from 0.8 is replaced with the polling API.

### CODE REVIEW

**Comment by rabbah** [comment]
Kafka test in Java failed in Travis. I suspect this has to do with the consumer group leader selection. Fixing.

**Comment by sjfink** [comment]
The comment "moves from a streaming interface to a polling interface":  tell me more?    what are the consequences?

**Comment by rabbah** [comment]
This is laying the ground work for a resource aware Kafka consumer with manual (explicit) offset management. We cannot do that with the streaming interface. The current change is "semantic preserving". The real change is coming next.

File: common/scala/src/whisk/connector/kafka/KafkaConsumerConnector.scala [hunk 1]
```diff
@@ -42,55 +42,60 @@ class KafkaConsumerConnector(
...
-                for (metadata <- stream) {
-                    val raw = new String(metadata.message, "utf-8")
-                    val msg = Message(raw)
-                    msg match {
-                        case Success(m) =>
-                            info(self, s"received /${metadata.topic}[${metadata.partition}][${metadata.offset}]: ${m.path}")(m.transid)
-                            process(m)
-                        case Failure(t) =>
-                            info(this, errorMsg(raw, t))
+                while (!disconnect) {
```
Comment (sjfink) [inline common/scala/src/whisk/connector/kafka/KafkaConsumerConnector.scala]: I'm wondering if the CPU usage of the new loop here is the same as the old loop.  ?

That would be possible if the old loop used a blocking interface to fetch messages from the stream, but the new implementation of getMessages() is non-blocking.      Not sure if this is the case -- but would like to understand.

File: tools/cli/wsk [hunk 2]
```diff
@@ -154,6 +154,8 @@ def propCmd(args, props, userprops, propsLocation):
             wskprop.updateProps('APIVERSION', args.apiversion, propsLocation)
             print 'ok: whisk API version set'
         if args.namespace:
+            if args.apihost is not None:
```
Comment (sjfink) [inline tools/cli/wsk]: This looks like a bug fix.   Do we have to ask users to download a new CLI?

Comment (rabbah) [inline tools/cli/wsk]: Not necessary. If the user tries to set their API host and namespace at the same time they'll get error (usually that their authentication is not valid). This fix corrects it but this is not common IMO.

File: common/scala/src/whisk/connector/kafka/KafkaConsumerConnector.scala (hunk 1 above)
Comment (rabbah) [inline common/scala/src/whisk/connector/kafka/KafkaConsumerConnector.scala]: They should be equivalent: stream returned as soon as at least one message is received. Poll is similar but will wait up to specified timeout. So poll is a long poll.

**Pull request merged**